### ISSUE TITLE
Implicit cast to ActionPreference in script does not work

### ISSUE THREAD

**Issue by mkht** [PowerShell/PowerShell#18776]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

First, open a console and execute the following commands one line at a time. This works as expected.

```PowerShell
PS C:\> echo $VerbosePreference
SilentlyContinue

PS C:\> $VerbosePreference.GetType().FullName
System.Management.Automation.ActionPreference

PS C:\> $VerbosePreference = 'Continue'

PS C:\> echo $VerbosePreference
Continue

PS C:\> $VerbosePreference.GetType().FullName
System.Management.Automation.ActionPreference
```

----
Second, save the exact same operation in a script file and execute it.
The type of VerbosePreference displayed in the execution result is different from that of the console execution.

If we assign a string value to `$VerbosePreference` in the console, it is implicitly cast to the ActionPreference enum, whereas if we do the same thing in the script, it appears not to be cast.

```powershell
# test script
echo $VerbosePreference
$VerbosePreference.GetType().FullName
$VerbosePreference = 'Continue'
echo $VerbosePreference
$VerbosePreference.GetType().FullName
```

```PowerShell
PS C:\> .\test.ps1
SilentlyContinue
System.Management.Automation.ActionPreference
Continue
System.String
```

### Expected behavior

```console
Even if we assign a string value to `$VerbosePreference` in the script, it should be implicitly cast to the ActionPreference enum type.
```

### Actual behavior

```console
it appears not to be cast.
```

### Environment data

```powershell
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iRon7** [comment]
I guess this ranks among general scoping behavior.
The reason for this is that `$VerbosePreference` doesn't actual exists in the local scope of the `.\test.ps1` script.
If you add `Get-Variable -Name VerbosePreference -Scope Local` to the top of the `.\test.ps1` script, and you will see an error:
```PowerShell
Get-Variable: G:\Mijn Drive\Scripts\PowerShell\Temp\ActionPreference.ps1:2
Line |
   2 |  Get-Variable -Name VerbosePreference -Scope Local
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find a variable with the name 'VerbosePreference'.
```
And if you invoke this command from the prompt, you will get the (global scoped) `$VerbosePreference`
```PowerShell
Name                           Value
----                           -----
VerbosePreference              SilentlyContinue
```
The point is that if you just do `echo $VerbosePreference` which only **reads** the value you will get the "inherited" value of the **global** `$VerbosePreference` value which appears to be explicitly declared as:
```PowerShell
[System.Management.Automation.ActionPreference]$VerbosePreference = 'SilentlyContinue'
```
This explains the type casting at the root (prompt).

Yet, if you try to **write** to a variable that is defined in a parent scope, PowerShell will simply create a **new** variable in the current scope which is declared according to your declaration (in your case a general object that contains a `String`).

In other words, I you would like to see the implicit casting of the (global) `$VerbosePreference` variable (the same as you accessing from the prompt), you should actually do this:

```PowerShell
$Global:VerbosePreference = 'Continue'
$Global:VerbosePreference
$Global:VerbosePreference.GetType().FullName
```

Which returns `System.Management.Automation.ActionPreference` in both cases.

**Comment by mkht** [comment]
Thank you. I did not fully understand the variable scope, but thanks to you I understand how it works.

My question has been resolved. I will close this issue.
